parlai/utils/torch.py

```diff
@@ -226,6 +226,29 @@ def argsort(keys: List[Any], *lists: List[List[Any]], descending: bool = False):
     return output
 
 
+def compute_grad_norm(parameters, norm_type=2.0):
+    """
+    Compute norm over gradients of model parameters.
+    Mostly taken from the body of torch.nn.utils.clip_grad_norm_
+
+    :param parameters:
+        the model parameters for gradient norm calculation. Iterable of
+        Tensors or single Tensor
+
+    :param norm_type
+        type of p-norm to use
+
+    """
+    if isinstance(parameters, torch.Tensor):
```

Comment:
I don't understand this one

Reply:
torch.nn.utils.clip_grad_norm_ works in the case where it's passed a list of tensors, as well as a single tensor of parameters. Since I don't think we'll be needing that case (?) I can just get rid of that

Reply:
Oh, that's cool. I like the mirroring, so you can leave it if you want.

Do you know what fixed the tests?